src/commands/git.rs

```diff
@@ -328,6 +323,44 @@ fn cmd_git_fetch(
     Ok(())
 }
 
+fn get_single_remote(git_repo: &git2::Repository) -> Result<Option<String>, CommandError> {
+    let git_remotes = git_repo.remotes()?;
+    Ok(match git_remotes.len() {
+        1 => git_remotes.get(0).map(ToOwned::to_owned),
+        _ => None,
+    })
+}
+
+const DEFAULT_REMOTE: &str = "origin";
```

Comment:
Do we need to treat a remote called "origin" differently? Could we instead say that we trust the user's config even if that is exactly "origin"? So as long as `config_remotes` is non-empty, use it.

Reply:
The thing is that the config has defaults for remotes and I wanted to preserve the current behavior and only do the autoselection thing if otherwise it would've failed.

If user has remotes "origin" and "other", currently jj would fetch origin without --remote, and if I remove the config default I'd still have to hardcode that config default to be returned here once we realize there's >1 remote.

Reply:
I've updated the code with the refactor I described to better illustrate what I meant and this approach is cleaner anyway - there is no two sources of the "origin" now

Reply:
> The thing is that the config has defaults for remotes and I wanted to preserve the current behavior and only do the autoselection thing if otherwise it would've failed.

Ah, I see. That makes sense. What do you think about removing the defaults from `misc.toml`? That way we can more easily detect if the user has not configured it. We could still default to "origin" if there are several remotes configured in the git repo and none configured in `git.push/fetch`. I would suggest even removing that default, but that can be a separate discussion in a separate PR.

Reply:
> removing the defaults from `misc.toml`? [..] still default to "origin" if there are several remotes

Well I did just that

Reply:
Oops, sorry I missed that! I'll review this now.